### PR TITLE
remove `etag` in response body to conform definition

### PR DESCRIPTION
**Pull request by msyyc**

### CODE REVIEW

File: packages/cadl-ranch-specs/http/azure/core/traits/main.tsp [hunk 1]
```diff
@@ -66,8 +66,7 @@ Expected response body:
 ```json
 {
   "id": 1,
-  "name": "Madge",
-  "etag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
+  "name": "Madge"
```
Comment (msyyc) [inline packages/cadl-ranch-specs/http/azure/core/traits/main.tsp:69]: `etag` is not defined in response body: https://github.com/Azure/cadl-ranch/blob/e91bb0289957f46a2b875b5cf9bf6b3a41d2bb6e/packages/cadl-ranch-specs/http/azure/core/traits/main.tsp#L41-L49

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/core/traits/main.tsp:69]: Should we improve scenario doc stating that SDK expect `etag` header and `bar` header in response (I assume they are)?

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/core/traits/main.tsp:69]: > `etag` is not defined in response body:
> 
> https://github.com/Azure/cadl-ranch/blob/e91bb0289957f46a2b875b5cf9bf6b3a41d2bb6e/packages/cadl-ranch-specs/http/azure/core/traits/main.tsp#L41-L49

The response does not need to define etag header explicitly, it is defined in the operation level with `SupportsConditionalRequests` trait.

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/core/traits/main.tsp:69]: I see. Sorry for my mis-understand. I agree to add doc to mention all response headers.

Comment (msyyc) [reply packages/cadl-ranch-specs/http/azure/core/traits/main.tsp:73]: doc is updated.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by tadelesh — APPROVED** [review]

**Pull request merged**